src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -626,6 +626,18 @@ namespace smt::noodler {
         conjuncts.push_back(conv_form_with_precision.first);
         precision = conv_form_with_precision.second;
 
+
+        // get the LIA formula describing solutions for special predicates
+        conjuncts.push_back(get_formula_for_ca_diseqs());
+        auto not_cont_prec = get_formula_for_not_contains();
+        if (not_cont_prec.second == LenNodePrecision::PRECISE || not_cont_prec.second == precision) {
+            conjuncts.push_back(not_cont_prec.first);
+        } else {
+            // if we should overwrite the precision, we instead return FALSE and say that we have underapproximation
+            conjuncts.push_back(LenNode(LenFormulaType::FALSE));
+            precision = LenNodePrecision::UNDERAPPROX;
```

Comment:
Is this correct? This will just say that it is unsat, but I think we should throw an error here instead.

Reply:
This basically says that we completely skip this branch of computation. If we returned `unsat` in the end, the underapproximation module outputs `unknown`. But there is still hope that we find solution in another branch.

Reply:
Ah, it is handled in `smt_tactic_core.cpp`, right?